HCollections/HListT.fs

```diff
@@ -7,7 +7,7 @@ open TypeEquality
 type HListT<'ts, 'elem> =
     private
     | Empty of Teq<'ts, unit>
-    | Cons of HListTConsCrate<'ts, 'elem>
+    | Cons of HListTConsCrate<'ts, 'elem> * length : int
```

Comment:
Can we do the same for `HListT`?

Reply:
Yes, but I'll do that in a different PR. This one is already getting a bit long, and your suggestion would involve adding to the public API before we really saw any benefit.